ground/gcs/src/plugins/boards_brainfpv/brainre1.cpp

```diff
@@ -77,6 +77,22 @@ bool BrainRE1::queryCapabilities(BoardCapabilities capability)
     case BOARD_CAPABILITIES_OSD:
     case BOARD_CAPABILITIES_UPGRADEABLE:
         return true;
+    case BOARD_CAPABILITIES_MAGS:
+        {
+            ExtensionSystem::PluginManager *pm = ExtensionSystem::PluginManager::instance();
+            UAVObjectManager *uavoManager = pm->getObject<UAVObjectManager>();
+            HwBrainRE1 *hwBrainRE1 = HwBrainRE1::GetInstance(uavoManager);
+            Q_ASSERT(hwBrainRE1);
+
+            if (!hwBrainRE1)
+                return false;
+
+            HwBrainRE1::DataFields settings = hwBrainRE1->getData();
+
+            if (settings.I2CExtMag != HwBrainRE1::I2CEXTMAG_NONE)
+                return true;
+        }
+        // Fall through.
     default:
```

Comment:
Existing code, but not really a fan of default case here, better to list them all explicitly so compilation breaks when a new capability is added, thus nothing gets missed (a very common problem!). Feel free to ignore this. :wink: